results.py

```diff
@@ -154,31 +163,41 @@ def __init__(
       fuzz_target_source: str = '',
       build_script_source: str = '',
       author: Any = None,
-      chat_history: Optional[dict] = None) -> None:
+      chat_history: Optional[dict] = None,
+      semantic_check: SemanticCheckResult = SemanticCheckResult(
+          SemanticCheckResult.NOT_APPLICABLE)
+  ) -> None:
     super().__init__(benchmark, trial, work_dirs, compiles, compile_error,
                      compile_log, binary_exists, is_function_referenced,
                      fuzz_target_source, build_script_source, author,
                      chat_history)
     self.crashes = crashes
     self.run_error = run_error
+    self.crash_func = crash_func or {}
     self.run_log = run_log
     self.coverage_summary = coverage_summary or {}
     self.coverage = coverage
     self.line_coverage_diff = line_coverage_diff
     self.reproducer_path = reproducer_path
+    self.artifact_path = artifact_path
+    self.artifact_name = artifact_name
+    self.sanitizer = sanitizer
     self.textcov_diff = textcov_diff
     self.log_path = log_path
     self.corpus_path = corpus_path
     self.coverage_report_path = coverage_report_path
     self.cov_pcs = cov_pcs
     self.total_pcs = total_pcs
+    self.semantic_check = semantic_check
```

Comment:
This should not be here unfortunately.
Maybe somewhere in `AnalysisResult`?
https://github.com/google/oss-fuzz-gen/blob/f08dd922b7fd2a2a1491fda36731e43c97d919ea/results.py#L225-L230

It can be `semantic_result` or `crash_result` or something new that describes the result of your agent better.